src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,147 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> currentNode = new Node<>(tail, value, null);
+            tail.next = currentNode;
+            tail = currentNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            Node<T> currentNode = new Node<>(null, value, head);
+            head.prev = currentNode;
+            head = currentNode;
+        } else {
+            Node<T> currentNode = nodeFromIndex(index);
+            Node<T> node = new Node<>(currentNode.prev, value, currentNode);
+            currentNode.prev.next = node;
+            currentNode.prev = node;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return nodeFromIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> currentNode = nodeFromIndex(index);
+        T returnValue = currentNode.value;
+        currentNode.value = value;
+        return returnValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> currentNode = nodeFromIndex(index);
+        return unLink(currentNode);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        if (object == null) {
+            while (currentNode != null) {
+                if (currentNode.value == null) {
+                    unLink(currentNode);
+                    return true;
+                }
+                currentNode = currentNode.next;
+            }
+        } else {
+            while (currentNode != null) {
+                if (currentNode.value.equals(object)) {
+                    unLink(currentNode);
+                    return true;
+                }
+                currentNode = currentNode.next;
+            }
```

Comment:
we dont need this "if-else" statement if we change your "if" statement inside the while loop like this:
if (object == node.item || (object != null && object.equals(node.item))) {
unlink(node);
return true;
}